honeybee/face.py

```diff
@@ -1,158 +1,875 @@
+        """
+        assert isinstance(door, Door), \
+            'Expected Door. Got {}.'.format(type(door))
+        self._acceptable_sub_face_check(Door)
+        door._parent = self
+        if self.normal.angle(door.normal) > math.pi / 2:  # reversed normal
+            door._geometry = door._geometry.flip()
+        self._door.append(door)
+        self._punched_geometry = None  # reset so that it can be re-computed
+
+    def add_apertures(self, apertures):
+        """Add a list of Apertures to this face."""
+        for aperture in apertures:
+            self.add_aperture(aperture)
+
+    def add_doors(self, doors):
+        """Add a list of Doors to this face."""
+        for door in doors:
+            self.add_door(door)
+
+    def set_adjacency(self, other_face, tolerance=0):
+        """Set this face adjacent to another and set the other face adjacent to this one.
+
+        Note that this method does not verify whether the other_face geometry is
+        co-planar or compatible with this one so it is recommended that either the
+        Face3D.is_centered_adjacent() or the Face3D.is_geometrically_equivalent()
+        method be used with this face geometry and the other_face geometry
+        before using this method in order to verify these criteria.
+
+        However, this method will use the proximity of apertures and doors within
+        the input tolerance to determine which of the sub faces in the other_face
+        are adjacent to the ones in this face. An exception will be thrown if not
+        all sub-faces can be matched.
+
+        Args:
+            other_face: Another Face object to be set adjacent to this one.
+            tolerance: The minimum distance between the center of two aperture
+                geometries at which they are condsidered adjacent. Default: 0.
+        """
+        # check the inputs and the ability of the faces to be adjacent
+        assert isinstance(other_face, Face), \
+            'Expected honeybee Face. Got {}.'.format(type(other_face))
+        assert len(self._apertures) == len(other_face._apertures), \
+            'Number of apertures does not match between {} and {}.'.format(
+                self.name, other_face.name)
+        assert len(self._doors) == len(other_face._doors), \
+            'Number of doors does not match between {} and {}.'.format(
+                self.name, other_face.name)
+        # set the boundary conditions of the faces
+        self._boundary_condition = Surface(other_face)
+        other_face._boundary_condition = Surface(self)
+        # set the apertures to be adjacent to one another
+        if len(self._apertures) > 0:
+            found_adjacencies = []
+            for aper_1 in self._apertures:
+                for aper_2 in other_face._apertures:
+                    if aper_1.center.distance_to_point(aper_2.center) <= tolerance:
+                        aper_1.set_adjacency(aper_2)
+                        found_adjacencies.append(True)
```

Comment:
why don't you just use a counter here instead of creating a list? As far as I can see the only use is to check the length of found adjacencies against the apertures.

Reply:
Thanks for catching this. I had the sense while writing this that it was not the correct way to do it but I had forgotten what the right way is.  I'll replace it with a counter now.